src/query/api/v1/handler/fetch_options.go

```diff
@@ -105,3 +136,81 @@ func newOrExistingRestrictFetchOptions(
 	}
 	return &storage.RestrictFetchOptions{}
 }
+
+// ParseStep parses the step duration for an HTTP request.
+func ParseStep(r *http.Request) (time.Duration, bool, error) {
+	step := r.FormValue(StepParam)
+	if step == "" {
+		return 0, false, nil
+	}
+	value, err := parseStep(r, StepParam)
+	if err != nil {
+		return 0, false, err
+	}
+	return value, true, err
+}
+
+func parseStep(r *http.Request, key string) (time.Duration, error) {
+	step, err := ParseDuration(r, key)
+	if err != nil {
+		return 0, err
+	}
+	if step <= 0 {
+		return 0, fmt.Errorf("expected postive step size, instead got: %d", step)
+	}
+	return step, nil
+}
+
+// ParseLookbackDuration parses a lookback duration for an HTTP request.
+func ParseLookbackDuration(r *http.Request) (time.Duration, bool, error) {
+	lookback := r.FormValue(LookbackParam)
+	if lookback == "" {
+		return 0, false, nil
+	}
+
+	if lookback == StepParam {
+		// Use the step size as lookback.
+		step, err := parseStep(r, StepParam)
```

Comment:
Maybe should explicitly add a sanity check for `step > 0`? It should fail already on `ParseStep` but sanity check here may be useful

Reply:
Yup so I added that inside `parseStep`

Reply:
Oops, I'm blind